File: src (cpp)/animal/reptilia/komodo.cpp
#include "komodo.h"

Komodo::Komodo(int _x, int _y, int _weight) : def_ID(10),
                                              def_ratio_meat(70),
                                              def_ratio_plant(0) {
  ID = def_ID;
  position.SetX(_x);
  position.SetY(_y);
  ratio_meat = def_ratio_meat;
  ratio_plant = def_ratio_plant;
  is_land_animal = true;
  is_water_animal = false;
  is_air_animal = false;
  weight = _weight;
}
void Komodo::Interact() {
  std::cout << "The komodo is looking at you intensely" << std::endl;
}

File: src/virtualzoo/animal/reptilia/Komodo.java
package virtualzoo.animal.reptilia;

import virtualzoo.animal.behavior.BehaviorTame;
import virtualzoo.animal.diet.Carnivora;

/**
 * Kelas Komodo mendefinisikan atribut untuk komodo.
 *
 * @author Rionaldi Chandraseta - 13515077
 * @version 3.0
 * @since 3.0
 */
public class Komodo extends Reptilia {

  /**
   * Constructor
   *
   * Menciptakan komodo dengan posisi (locX,locY) dan berat animalWeight.
   *
   * @param locX Nilai absis posisi.
   * @param locY Nilai ordinat posisi.
   * @param animalWeight Nilai berat iguana.
   */
  public Komodo(int locX, int locY, int animalWeight) {
    super();
    animalBehavior = new BehaviorTame();
    animalDiet = new Carnivora(animalWeight, 70);
    ID = 10;
    position.setLocation(locX, locY);
    isLandAnimal = true;
    isWaterAnimal = false;
    isAirAnimal = false;
  }

  /**
   * Mengatur interaksi dengan komodo.
   */
  public void interact() {
    System.out.println("The komodo is looking at you intensely");
  }
}
